src/telegram_fmt_plug.c

```diff
@@ -52,11 +53,15 @@ john_register_one(&fmt_telegram);
 #define SALT_SIZE               sizeof(struct custom_salt)
 #define SALT_ALIGN              sizeof(uint64_t)
 #ifdef SIMD_COEF_32
-#define MIN_KEYS_PER_CRYPT      SSE_GROUP_SZ_SHA1
-#define MAX_KEYS_PER_CRYPT      (SSE_GROUP_SZ_SHA1 * 4)
+#define SHA1_LOOP_CNT            (SIMD_COEF_32 * SIMD_PARA_SHA1)
+#define SHA512_LOOP_CNT          (SIMD_COEF_64 * SIMD_PARA_SHA512)
+#define MIN_KEYS_PER_CRYPT       (SIMD_COEF_32 * SIMD_PARA_SHA1 * SIMD_PARA_SHA512)
+#define MAX_KEYS_PER_CRYPT       (SIMD_COEF_32 * SIMD_PARA_SHA1 * SIMD_PARA_SHA512)
```

Comment:
Looks like these and `ALGORITHM_NAME` assume that if we have `SIMD_COEF_32` then we also have `SIMD_COEF_64`. IIRC, this assumption is false for some older POWER architecture revision, which only had 32-bit SIMD vector components.

Also, `SIMD_PARA_SHA1 * SIMD_PARA_SHA512` may be excessive for any one of these two format variations. I see why we have to use it here - our `min_keys_per_crypt` is global, but we may have loaded both kinds of "hashes" at once, yet we need our loops in `crypt_all()` to use exact multiples.

To me, these are some reasons to split this format in two.

With this being just one format, we'd also need to report the two different versions as "tunable costs" to be able to benchmark them separately, and even then our default `--test` will either benchmark only one or it will benchmark a weird mix of them rather than the two separately. This is yet another reason for the split, or alternatively we need to enhance the formats API to let the benchmarking code run separate benchmarks per format subtype, and then enhance that code to actually do that.